app/src/main/java/com/github/featuredetectandroid/utils/PreferencesManager.kt

```diff
@@ -0,0 +1,25 @@
+package com.github.featuredetectandroid.utils
+
+import android.content.Context
+import android.util.Log
+import androidx.core.content.edit
+import com.github.featuredetectandroid.MainActivity
+
+private const val TAG = "PreferencesManager"
+private const val KEY = "algorithm"
+private const val NONE = "None"
+
+class PreferencesManager(mainActivity: MainActivity) {
+    private val preferences = mainActivity.getPreferences(Context.MODE_PRIVATE)
+
+    fun getAlgorithm() = preferences.getString(KEY, NONE) ?: NONE
+
+    fun selectAlgorithm(newAlgorithm: String) {
```

Comment:
Let's name it `putSelectedAlgorithm` because if there'll be more keys figuring out special verbs like this for them will be a pain

Reply:
fixed

Reply:
fixed